pkg/workos_errors/authentication_errors.go

```diff
@@ -0,0 +1,204 @@
+	PendingAuthenticationToken string   `json:"pending_authentication_token"`
+}
+
+func (e SSORequiredError) Error() string {
+	return e.HTTPError.Error()
+}
+
+// OrganizationAuthenticationMethodsRequiredError occurs when org restricts auth methods
+type OrganizationAuthenticationMethodsRequiredError struct {
+	HTTPError
+	ErrorCode                  string          `json:"error"`
+	ErrorDescription           string          `json:"error_description"`
+	Email                      string          `json:"email"`
+	SSOConnectionIDs           []string        `json:"sso_connection_ids"`
+	AuthMethods                map[string]bool `json:"auth_methods"`
+	PendingAuthenticationToken string          `json:"pending_authentication_token"`
+}
+
+func (e OrganizationAuthenticationMethodsRequiredError) Error() string {
+	return e.HTTPError.Error()
+}
+
+// Type checking functions
+func IsEmailVerificationRequired(err error) bool {
+	var emailErr *EmailVerificationRequiredError
+	return errors.As(err, &emailErr)
+}
+
+func IsMFAEnrollment(err error) bool {
+	var mfaErr *MFAEnrollmentError
+	return errors.As(err, &mfaErr)
+}
+
+func IsMFAChallenge(err error) bool {
+	var mfaErr *MFAChallengeError
+	return errors.As(err, &mfaErr)
+}
+
+func IsOrganizationSelectionRequired(err error) bool {
+	var orgErr *OrganizationSelectionRequiredError
+	return errors.As(err, &orgErr)
+}
+
+func IsSSORequired(err error) bool {
+	var ssoErr *SSORequiredError
+	return errors.As(err, &ssoErr)
+}
+
+func IsOrganizationAuthenticationMethodsRequired(err error) bool {
+	var orgAuthErr *OrganizationAuthenticationMethodsRequiredError
+	return errors.As(err, &orgAuthErr)
+}
+
+// Type assertion functions
+func AsEmailVerificationRequired(err error) (*EmailVerificationRequiredError, bool) {
+	var emailErr *EmailVerificationRequiredError
+	if errors.As(err, &emailErr) {
+		return emailErr, true
+	}
+	return nil, false
+}
+
+func AsMFAEnrollment(err error) (*MFAEnrollmentError, bool) {
+	var mfaErr *MFAEnrollmentError
+	if errors.As(err, &mfaErr) {
+		return mfaErr, true
+	}
+	return nil, false
+}
+
+func AsMFAChallenge(err error) (*MFAChallengeError, bool) {
+	var mfaErr *MFAChallengeError
+	if errors.As(err, &mfaErr) {
+		return mfaErr, true
+	}
+	return nil, false
+}
+
+func AsOrganizationSelectionRequired(err error) (*OrganizationSelectionRequiredError, bool) {
+	var orgErr *OrganizationSelectionRequiredError
+	if errors.As(err, &orgErr) {
+		return orgErr, true
+	}
+	return nil, false
+}
+
+func AsSSORequired(err error) (*SSORequiredError, bool) {
+	var ssoErr *SSORequiredError
+	if errors.As(err, &ssoErr) {
+		return ssoErr, true
+	}
+	return nil, false
+}
+
+func AsOrganizationAuthenticationMethodsRequired(err error) (*OrganizationAuthenticationMethodsRequiredError, bool) {
+	var orgAuthErr *OrganizationAuthenticationMethodsRequiredError
+	if errors.As(err, &orgAuthErr) {
+		return orgAuthErr, true
+	}
+	return nil, false
+}
```

Comment:
Is this something we usually do for errors? Do we have a usage example?

Reply:
After additional thought I wonder if we could go without these `Is*` and `As*` functions for now. Perhaps it's a bit overkill, at least for now, unless developers have said otherwise.

We could provide the error types and then the developer would do a type assertion as such in their code:
```
err := client.AuthenticateUser(ctx, opts)
var emailVerificationErr *EmailVerificationRequiredError

if errors.As(err, &emailVerificationErr) {
    // Access specific fields: emailErr.Email, emailErr.PendingAuthenticationToken, etc.
}
```

Reply:
Refactored to no longer use these helper functions and only provide the types for the developers to use with the errors.As usage pattern.